dozer/cogs/management.py

```diff
@@ -0,0 +1,209 @@
+timezone_file = "timezones.json"
+
+
+class Management(Cog):
+    """A cog housing Guild management/utility commands."""
+
+    def __init__(self, bot):
+        super().__init__(bot)
+        self.started_timers = False
+        self.timers = {}
+        if os.path.isfile(timezone_file):
+            DOZER_LOGGER.info("Loaded timezone configurations")
+            with open(timezone_file) as f:
+                self.timezones = json.load(f)
+        else:
+            DOZER_LOGGER.error("Unable to load timezone configurations")
+            self.timezones = {"abbreviations": []}
+
+    @Cog.listener('on_ready')
+    async def on_ready(self):
+        """Restore time based event schedulers"""
+        messages = await ScheduledMessages.get_by()
+        started = 0
+        if not self.started_timers:
+            for message in messages:
+                task = self.bot.loop.create_task(self.msg_timer(message))
+                self.timers[message.request_id] = task
+                started += 1
+            self.started_timers = True
+            DOZER_LOGGER.info(f"Started {started}/{len(messages)} scheduled messages")
+        else:
+            DOZER_LOGGER.info("Client Resumed: Timers still running")
+
+    async def msg_timer(self, db_entry):
+        """Holds the futures for sending a message"""
+        delay = db_entry.time - datetime.now(tz=timezone.utc)
+        if delay.total_seconds() > 0:
+            await asyncio.sleep(delay.total_seconds())
+        await self.send_scheduled_msg(db_entry)
+        await db_entry.delete(request_id=db_entry.request_id)
+
+    async def send_scheduled_msg(self, db_entry, channel_override=None):
+        """Formats and sends scheduled message"""
+        embed = discord.Embed(title=db_entry.header if db_entry.header else "Scheduled Message", description=db_entry.content)
+        guild = self.bot.get_guild(db_entry.guild_id)
+        channel = guild.get_channel(db_entry.channel_id if not channel_override else channel_override)
+        embed.colour = blurple
+        if db_entry.requester_id:
+            name = await guild.fetch_member(db_entry.requester_id)
+            embed.set_footer(text=f"Author: {name.display_name}")
+        await channel.send(embed=embed)
+
+    @group(invoke_without_command=True)
+    @has_permissions(manage_messages=True)
+    async def schedulesend(self, ctx):
+        """Allows a message to be sent at a particular time
+        Commands: add, delete, list
+        """
+
+    @schedulesend.command()
+    @has_permissions(manage_messages=True)
+    async def add(self, ctx, channel: discord.TextChannel, time, *, content):
+        """Allows a message to be sent at a particular time
+        Headers are distinguished by the characters `-/-`
+        """
+        print(self.timezones)
+        send_time = parser.parse(time, tzinfos=self.timezones['abbreviations'])
+        content = content.split("-/-", 1)
+        message = content[1] if len(content) == 2 else content[0]
```

Comment:
Both the message and header should be checked for message length here such that the embed can be created properly.